recipes-nymea/nymea-plugins/nymea-plugins_git.bb

```diff
@@ -49,4 +49,4 @@ ALLOW_EMPTY_${PN} = "1"
 FILES_${PN} = ""
 
 # Dynamically generate packages for all enabled plugins
-PACKAGES_DYNAMIC = "^nymea-plugin-.*"
+PACKAGES_DYNAMIC = "^nymea-plugin-(?!zigbee).*"
```

Comment:
not sure what warrants this change? what are you trying to do here?

Reply:
So how understood it is: We have now two plugin repos, one without zigbee plugins (nymea-plugins) and one with only (nymea-plugins-zigbee).

I created the second recipe to provide the zigbee plugins in the same fashion as the non-zigbee plugins, just copied the recipe. However, now I think what is happening is that `nymea-plugins_git.bb` can not provide e.g. `nymea-plugin-zigbeegeneric` although `PACKAGES_DYNAMIC` does indicate that it would.

And thats why I run into:
```
ERROR: yocto-.... do_rootfs: Unable to install packages. Command '...' returned 255:
Collected errors:
 * opkg_prepare_url_for_install: Couldn't find anything to satisfy 'nymea-plugin-zigbeegeneric'.
``´

Although recipe `nymea-plugins-zigbee_git.bb` would indeed provide these packages.
So my way of working around this was to specify that this recipe provides everything except zigbee plugins. And then the zigbee recipe provides all zigbee ones.

Not sure if this can be done in a better way.

Reply:
> So how understood it is: We have now two plugin repos, one without zigbee plugins (nymea-plugins) and one with only (nymea-plugins-zigbee).
> 
> I created the second recipe to provide the zigbee plugins in the same fashion as the non-zigbee plugins, just copied the recipe. However, now I think what is happening is that `nymea-plugins_git.bb` can not provide e.g. `nymea-plugin-zigbeegeneric` although `PACKAGES_DYNAMIC` does indicate that it would.
> 
> And thats why I run into:
> 
> ```
> ERROR: yocto-.... do_rootfs: Unable to install packages. Command '...' returned 255:
> Collected errors:
>  * opkg_prepare_url_for_install: Couldn't find anything to satisfy 'nymea-plugin-zigbeegeneric'.
> ```
> 
> Although recipe `nymea-plugins-zigbee_git.bb` would indeed provide these packages. So my way of working around this was to specify that this recipe provides everything except zigbee plugins. And then the zigbee recipe provides all zigbee ones.
> 
> Not sure if this can be done in a better way.

Oooooh. New stuff to dig into for me as I was not aware of this mechanism! Thanks for explaining (would be a great commit log and/or comment!). You could rename the nymea-plugins plugins into something like nymea-plugin-core- or whatever suits best so that there is no naming collision.

It's a bit confusing though to have nymea-plugins and nymea-plugins-zigbee.. @mzanetti what's the rationale behind those additional git repos, seemingly storing the same thing - plugins? Is it akin to gstreamer good/bad/ugly plugins system? nymea-plugins only has first citizen plugins and then other git repos are here until the plugins are deemed stable enough to be merged into nymea-plugins?

Otherwise, I don't see another solution unfortunately (except not using the dynamic package feature which I personally really like for not having to maintain a list of PACKAGES in the recipe, which makes updates that much easier to deal with).

> Just to add some information: There are even more plugin repositories (nymea-plugins-genericthings, nymea-plugins-modbus etc...). Only the main plugins repository and the zigbee one are currently of interest to this particular project tho but this "problem" will arise more often in the future as more different plugin repositories are added.
> 
> My (somewhat underqualified) idea was to perhaps generate this from the PACKAGECONFIG which should have a list of all the built plugins of a particular repository/recipe?

That could be done but requires more active maintenance. You would need to maintain the list of PACKAGECONFIG in sync with the update, which hasn't been done in the past and is time consuming and error prone. The nice thing about the current mechanism is that any release with more/less plugins will just build the packages without any more changes than the SRCREV, create a package for it and if the user is using nymea-plugins-zigbee meta package in their image, it'll bring those new plugin packages in without any additional effort.

Reply:
As we don't really have a good alternative way of doing this I decided to keep it as it is. We will have to add new repos/plugin groups when needed, however not maintaining any kind of list of plugins is preferred.